src/core.js

```diff
@@ -0,0 +1,76 @@
+import { unlerp } from '@kaliber/math'
+
+function calculateScrollProgression({ element, container, start, end }) {
+  const scrollPosition = container === window ? window.scrollY : container.scrollTop
+  return unlerp({
+    start: calculatePosition({ element, container, position: start, scrollPosition }),
+    end: calculatePosition({ element, container, position: end, scrollPosition }),
+    clamp: true,
+    input: scrollPosition
+  })
+}
+
+function calculatePosition({ element, container, position, scrollPosition }) {
+  const elementRect = element.getBoundingClientRect()
+  const containerRect = container === window ? getWindowRect() : container.getBoundingClientRect()
```

Comment:
Op verschillende plekken maak je het onderscheid tussen `window` en 'iets anders'. Het is daarom geen gek idee om zoiets te doen:

```js
container === window ? windowContainer() : elementContainer(container)

...

function windowContainer() {
  return {
    get rect() { ... },
    get scrollPosition() { ... },
    get canScroll() { ... },
  }
}

function elementContainer(element) {
  return {
    get rect() { ... },
    get scrollPosition() { ... },
    get canScroll() { ... },
  }
}
```

Dan hoef je in de rest van je code geen if statements meer te gebruiken.

Reply:
Ah, jep. Nice :)